### PR TITLE
overlord/snapstate: introduce envvars to control the channels for bases and prereqs

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: overlord/snapstate/handlers.go [hunk 1]
```diff
@@ -113,7 +113,22 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 */
 
 const defaultCoreSnapName = "core"
-const defaultBaseSnapsChannel = "stable"
+
+func defaultBaseSnapsChannel() string {
+	channel := os.Getenv("SNAPD_BASES_CHANNEL")
```
Comment (zyga) [inline overlord/snapstate/handlers.go]: Could we document those someplace. This will be forgotten next week.

Comment (pedronis) [reply overlord/snapstate/handlers.go]: they are for our own testing tough, not for users

Comment (zyga) [reply overlord/snapstate/handlers.go]: That doesn't preclude those from being documented. It's a hidden feature really.

Comment (pedronis) [reply overlord/snapstate/handlers.go]: We don't have a place atm where we document this stuff (I only see SNAPD_DEBUG documented in HACKING.md and that's it) , they seem the wrong ones to start if we document only these.

Comment (zyga) [reply overlord/snapstate/handlers.go]: Traditionally those are documented on the manual page of the command that respects / uses them. I did this for snap-confine but we have a pretty poor manual page for snap/snapd for now.

**Pull request merged**